### PR TITLE
Expose darwinssl as an option when building curl

### PR DESCRIPTION
**Pull request by sgr-ms**
This exposes the ability to built curl while using Apple's SSL/TLS implementation.

### CODE REVIEW

File: ports/curl/portfile.cmake [hunk 1]
```diff
@@ -87,6 +98,7 @@ vcpkg_configure_cmake(
         -DCMAKE_USE_OPENSSL=${USE_OPENSSL}
         -DCMAKE_USE_WINSSL=${USE_WINSSL}
         -DCMAKE_USE_MBEDTLS=${USE_MBEDTLS}
+        -DCMAKE_USE_DARWINSSL=$USE_DARWINSSL}
```
Comment (past-due) [inline ports/curl/portfile.cmake]: Looks like a small typo here? `$USE_DARWINSSL}` -> `${USE_DARWINSSL}`

**Comment by vicroms** [comment]
CI doesn't test non-default features. 
@Rastaban can you test installing `curl[darwinssl]` locally?

**Comment by sgr-ms** [comment]
> 
> 
> CI doesn't test non-default features.
> @Rastaban can you test installing `curl[darwinssl]` locally?
> 
> @sgr-ms
> As this only works for Mac, I wonder if adding a validation to disable the feature for non-Mac should be added. Although, it would be wrong for a user to attempt installing the feature on any other platform.
> 
> `if("darwinssl" IN_LIST FEATURES AND VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")`

Sure, I have added something similar to what winssl has.

**Comment by Rastaban** [comment]
@vicroms sure, I will test it out later today after I finish setting up my MacBook.

**Comment by past-due** [comment]
@Rastaban:

It would be nice if we could update the `curl[ssl]` feature to default to `darwinssl` on Apple platforms.

So:
```
Feature: ssl
Build-Depends: curl[openssl] (!windows), curl[winssl] (windows)
Description: Default SSL backend
```

could become something like:
```
Feature: ssl
Build-Depends: curl[openssl] (!windows&!macos), curl[winssl] (windows), curl[darwinssl] (macos)
Description: Default SSL backend
```

But is it currently possible to specify `macos`/`darwin` to filter only Apple platforms in the `CONTROL` file?

**Comment by Rastaban** [comment]
@past-due you can filter on Mac by using the name `osx`.  See the antlr4 CONTROL file as an example.

**Comment by sgr-ms** [comment]
> this



> 
> 
> @past-due you can filter on Mac by using the name `osx`. See the antlr4 CONTROL file as an example.

What about if you also use this to build for ios (like me) ? Do I need to specify each OS (e.g. iphoneos and simulator) as well ?

**Comment by Rastaban** [comment]
Currently that filtering is based on the name of the triplet file (although that may change sometime in the future).  So yes, unless your ios triplet file also has `osx` in its name it will also need to be listed.

**Comment by sgr-ms** [comment]
> 
> 
> Currently that filtering is based on the name of the triplet file (although that may change sometime in the future). So yes, unless your ios triplet file also has `osx` in its name it will also need to be listed.
> 
> For this PR I am okay with only taking `osx` into account in the curl[ssh] feature. If you build curl[ssh] for iphones you would get the existing behavior by default, if you want darwinssl then you could either build curl[darwinssl] explicitly or modify the local copy of the control file to take your iphone triplet into account. Official support for ios is on our backlog (although it keeps getting pushed back) when that happens I expect we would make changes to this and other similar ports.

@Rastaban Seems like this broke the CI verification ? Or is this unrelated?

**Comment by Rastaban** [comment]
Rerunning CI validation, the failing test is from a flakey port.

**Comment by Rastaban** [comment]
it looks like google-cloud-cpp:x64-osx is failing.  I won't be able to get to the logs for this until Monday.

**Comment by past-due** [comment]
@Rastaban: PR #5968 should fix the breakage in `google-cloud-cpp:x64-osx` caused by this port.

**Comment by past-due** [comment]
@sgr-ms: Can you rebase this branch on the latest master? (With PR #5968 and PR #5971 merged, the `google-cloud-cpp:x64-osx` issues should be fixed.)

**Comment by sgr-ms** [comment]
@past-due Is there anything else I can do to merge this ?

**Comment by vicroms** [comment]
I'm not sure what happened with this PR's git history, but it has become quite hard to review.

**Comment by sgr-ms** [comment]
@vicroms 
I rebased it on the latest master at the time, maybe something went wrong there ?

**Comment by past-due** [comment]
The pertinent commits should be:
095ce4e
1d8d9ad
6af7229
4ae524d

(Perhaps it was the merge commit that added all the latest master commits to the history of this PR?)

**Comment by sgr-ms** [comment]
> 
> 
> The pertinent commits should be:
> [095ce4e](https://github.com/Microsoft/vcpkg/commit/095ce4e9392a83a2e4944635c9f0e2b43caa0fe9)
> [1d8d9ad](https://github.com/Microsoft/vcpkg/commit/1d8d9ad35a24e56c6ac8e4b8deb567019a64a5f8)
> [6af7229](https://github.com/Microsoft/vcpkg/commit/6af72299f3e0d9efc53596caddbd8ed04e6477f8)
> [4ae524d](https://github.com/Microsoft/vcpkg/commit/4ae524d7962b8253115c222d69c1f7bc3294f998)
> 
> (Perhaps it was the merge commit that added all the latest master commits to the history of this PR?)

Probably :\
Is there anything I can do to fix this ?

**Comment by past-due** [comment]
I cherry-picked just the pertinent commits, and opened a new PR here: #6124 (to make it easier to review).

**Pull request closed without merging**